configure.ac

```diff
@@ -250,6 +250,19 @@ if test "x$GCC" = "xyes" ; then
   JE_CFLAGS_ADD([-Wsign-compare])
   JE_CFLAGS_ADD([-Wundef])
   JE_CFLAGS_ADD([-Wno-format-zero-length])
+  dnl For conditionally compiled features like profiling, we always include
+  dnl string buffers, even when the feature is disabled, in order to prevent us
+  dnl having to ifdef-away all the places where it touched (even though those
+  dnl places get compiled away).  To minimize the space overhead, we change the
+  dnl lengths to 1 when the feature is disabled.  Since we occasionally strncpy
+  dnl with a size of len - 1, this can lead to a call to strncpy with a size of
+  dnl 0, which triggers a string warning.
+  JE_CFLAGS_ADD([-Wno-stringop-truncation])
```

Comment:
For the prof related string ops, maybe introduce a prof strncpy wrapper which bails when prof is off? I’m slightly hesitant here.

Reply:
I don't think this suffices to fix the warning: https://gcc.godbolt.org/z/cm3ygz .

Reply:
Yeah I remember now... let’s guard the strncpy with ifdef? Assuming it’s only one place, it feels a bit better than turning the check off.

Reply:
Done.